### PR TITLE
Simplify prune algorithm

### PR DESCRIPTION
**Pull request by Isaac-DeFrain**
- simplifies root branch pruning algorithm and ensures forks are dealt with appropriately
- adds `--prune_interval` cli flag
- adds `PartialOrd` and `Ord` impls for `Block`
- modifies tests to use sequential blocks over beautified
- removes the beautified sequential blocks

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204820932108695

### CODE REVIEW

File: src/state/mod.rs [hunk 1]
```diff
@@ -106,14 +101,10 @@ impl IndexerState {
...
         if let Some(k) = self.transition_frontier_length {
-            if let Some(interval) = self.prune_interval {
-                if self.blocks_processed % interval == 0 {
-                    info!("pruning transition frontier at k={}", k);
-                    self.root_branch.prune_transition_frontier(k);
-                }
-            } else {
-                info!("pruning transition frontier at k={}", k);
-                self.root_branch.prune_transition_frontier(k);
+            if self.root_branch.height() > 5 * k as usize {
```
Comment (jenr24G) [inline src/state/mod.rs:104]: can we make this constant configurable?

**Review by jenr24G — APPROVED** [review]
LGTM! I'd like to add a configuration option for how often we prune the root branch. Different use cases may warrant different pruning rates!

https://app.asana.com/0/1203669416123378/1204820393125452/f

**Review by trevorbernard — APPROVED** [review]
:shipit:

File: src/state/mod.rs (hunk 1 above)
Comment (Isaac-DeFrain) [reply src/state/mod.rs:104]: I think that's a good idea!

Comment (Isaac-DeFrain) [reply src/state/mod.rs:104]: Will include with my fix for updating the best tip

**Review by trevorbernard — APPROVED** [review]
:shipit:

**Pull request merged**